tests/test_sailor/test_assetcentral/test_utils.py

```diff
@@ -225,3 +227,71 @@ def test_apply_filters_post_request_property_mapping():
     actual = _apply_filters_post_request(data, equality_filters, extended_filters, property_mapping)
 
     assert actual == expected_result
+
+
+@pytest.mark.parametrize('testdescription,unbreakable_filters,breakable_filters,expected', [
+    ('empty input returns empty filters',
+        [], [], []),
+    ('unbreakable filters are combined with "and"',
+        ["name eq 'test'", "location eq 'Paris'"], [],
+        ["name eq 'test' and location eq 'Paris'"]),
+    ('breakable filters are combined with "or"',
+        [], [["location eq 'Paris'", "location eq 'London'"]],
+        ["(location eq 'Paris' or location eq 'London')"]),
+    ('multiple breakable filters are connected with "and"',
+        [], [["testFac eq 'abcCorp'", "testFac eq '123pumps'"], ["location eq 'Paris'", "location eq 'London'"]],
+        ["(testFac eq 'abcCorp' or testFac eq '123pumps') and (location eq 'Paris' or location eq 'London')"]),
+    ('un- and breakable filters are connected with "and"',
+        ["name eq 'test'"], [["location eq 'Paris'", "location eq 'London'"]],
+        ["name eq 'test' and (location eq 'Paris' or location eq 'London')"])
+])
+def test_compose_queries(unbreakable_filters, breakable_filters, expected, testdescription):
+    actual = _compose_queries(unbreakable_filters, breakable_filters)
+    assert actual == expected
+
+
+def test_compose_queries_too_many_filters_are_split():
+    # the correctness of the test depends on what is configured as the max_filter_length in _compose_queries
+    unbreakable_filters = []
+    breakable_filters = [["manufacturer eq 'abcCorp'", "manufacturer eq '123pumps'"] * 50,
```

Comment:
maybe that test would be even stronger if the names of the manufacturer would vary (simply adding an `f'abcCorp_{i}'`/`f'123pumps_{i}'` in a list comprehension rather than the multiplication you have now.
Then of course the expected filter start would need to use the right numbers for the second filter, which would make it clear the filters are not just repeated :)
(maybe even testing that the last entries in the first filter and the first entries in the second filter align)